pkg/scheduler/factory.go

```diff
@@ -343,10 +343,13 @@ func MakeDefaultErrorFunc(client clientset.Interface, podLister corelisters.PodL
 			klog.Warningf("Pod %v/%v doesn't exist in informer cache: %v", pod.Namespace, pod.Name, err)
 			return
 		}
-		// As <cachedPod> is from SharedInformer, we need to do a DeepCopy() here.
-		podInfo.Pod = cachedPod.DeepCopy()
-		if err := podQueue.AddUnschedulableIfNotPresent(podInfo, podQueue.SchedulingCycle()); err != nil {
-			klog.Error(err)
+
+		if len(pod.Spec.NodeName) == 0 {
```

Comment:
Let's use `cachedPod`:

```go
// Somehow the live Pod may have been assigned a node name.
if len(cachedPod.Spec.NodeName) != 0 {
	klog.Warningf("Pod %v/%v has been assigned with %v. Abort adding it back to queue.", pod.Namespace, pod.Name, cachedPod.Spec.NodeName)
	return
}
```

Reply:
Yes, you're right. I'll fix this problem today and describe the scenario where it happened.